### DISCUSSION TITLE
Adding custom resources to the docker images, to enable use by all sites

### DISCUSSION THREAD

**Discussion by markwturnbull** [squidfunk/mkdocs-material#7377] [Q&A] [answered]
Hello Community 

I have a custom docker image created around mkdocs-material 9.3.2, that is used by multiple site's to do the build.  

To simplify site configuration, additional images & resources that are used by all sites, have been added into the `lib/python3.11/site-packages/material` folder, in a sub folder named 'custom-assets'.  

I know this may not be the way mkdocs-material was envisioned to work, but it allowed common resources (images, js, favicon) to be centralized, and simply referenced by multiple sites. A point to note is that these resources are not available from an external web browser.

Following an upgrade of mkdocs-material, this functionality has stopped working.  The custom-assets folder is no longer available to the site, either in build or serve mode.

Tracing the change back, this started with 9.4.0.

Is this a bug in mkdocs-material? Or is there a better way (without hosting on a webserver) to achieve the same results?

**Comment by kamilkrzyskow** [comment, marked as answer]
Hi @markwturnbull,
I'm not sure what are the files you have in the `custom-assets` directory, and how you're accessing them.
But thanks for the pointer that it changed in 9.4.0.
In that version the theme entry point has changed. 
Instead of the root of the `material` package the theme is inside the `templates` directory/sub-package.

https://github.com/squidfunk/mkdocs-material/tree/9.3.2/material
https://github.com/squidfunk/mkdocs-material/tree/9.4.0/material

https://github.com/squidfunk/mkdocs-material/blob/d848c46ffacf946f445d978d1b3c9c768d3ffa45/pyproject.toml#L83-L84
https://github.com/squidfunk/mkdocs-material/blob/8b44f1a74d0d74da7273d48b9cd791a74162bfea/pyproject.toml#L83-L84

**Reply by markwturnbull** [reply]
Hi @kamilkrzyskow 

Thank you very much :)

That solved my issue :)

**Reply by kamilkrzyskow** [reply]
Would you perhaps mind sharing your setup for extending the docker image @markwturnbull?
I think some people would like to do it themselves, but perhaps don't know how 🤔
